cmd/ipfs/daemon.go

```diff
@@ -109,6 +113,14 @@ func daemonFunc(req cmds.Request, res cmds.Response) {
 		}
 	}()
 
+	// check transport encryption flag.
+	unencrypted, _, _ := req.Option(unencryptTransportKwd).Bool()
+	if unencrypted {
+		log.Warningf(`Running with --%s: All connections are UNENCRYPTED.
+		You will not be able to connect to regular encrypted networks.`, unencryptTransportKwd)
+		conn.EncryptConnections = false
```

Comment:
I just want to guard (package) users from using this in unintended ways and make it clear to them that this is a development feature. A func could disallow you to switch encryption back on. 

The rest is fine by me. Of course we can't do anything meaningful about corruption from the outside.